hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystem.java

```diff
@@ -2019,4 +2019,9 @@ public boolean hasPathCapability(final Path path, final String capability)
   public IOStatistics getIOStatistics() {
     return abfsCounters != null ? abfsCounters.getIOStatistics() : null;
   }
+
+  @VisibleForTesting
+  void setAbfsStore(AzureBlobFileSystemStore store) {
+    this.abfsStore = store;
+  }
```

Comment:
please avoid setters.
what can be done is , we can have a spied fs.
```
AzureBlobFileSystemStore store = Mockito.spy(fileSystem.getAbfsStore());
Mockito.doReturn(store).when(fileSystem).getAbfsStore();
```

Reply:
In this case, only having a spied fs will not help as we need to mock the client and store behavior and they are set up during the init phase, and not using the getAbfsStore() method.